src/Sulu/Bundle/TestBundle/Testing/ContainerCompatibilityTrait.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\TestBundle\Testing;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+trait ContainerCompatibilityTrait
+{
+    /**
+     * @internal The "getContainer" function is deprecated use self::$container variable instead.
+     *
+     * This function exist to keep a compatibility between symfony versions.
```

Comment:
So as soon as Symfony 3.4 is not supported anymore, we can delete this trait? Then I would mention that here.

And why is that the case? Does that mean we always have to use this function now?

Btw, what's the state of Symfony 5 support? Can we support 3.4 at the same time? Wasn't that the condition to keep the 3.4 support?

Reply:
5.0 will definitly need more work as this. As there are some other changes in the browser kit and http kernel.